samples/cpp/benchmark_app/main.cpp

```diff
@@ -402,6 +402,8 @@ int main(int argc, char* argv[]) {
                 setThroughputStreams();
             } else if (device.find("GNA") != std::string::npos) {
                 set_infer_precision();
+            } else if (device.find("AUTO") != std::string::npos) {
```

Comment:
@myshevts AUTO doesn't support streams itself, is not it? If command line contain -nstreams it would be applied to all devices, including AUTO

Reply:
> @myshevts AUTO doesn't support streams itself, is not it? If command line contain -nstreams it would be applied to all devices, including AUTO

I see but previous beh was to throw excpetion on the unsupported config, rather than swallow (ignore) the unaaplicable user's setting.... also having if (AUTO) else if (..) is not future proof. If we want to ignore the config option, let's just check that the device doesn't support that

Reply:
other wise we will keep adding if (BATCH) if CUDA if ???

Reply:
@myshevts It make sense. But AUTO/MULTI/HETERO anyway has special handling here. And unfortunately we still have if-else branches for CPU/GPU/MYRIAD in main.cpp